### PR TITLE
test(symbolication): Add tests for DIF candidates

### PR DESCRIPTION
**Pull request by flub**
This adds an integration test which checks DIF candidates can be
requested from symbolicator.  The test uses a DIF on the sentry:project
source which also asserts the custom URI scheme for these which the UI
relies on.

### CODE REVIEW

**Comment by flub** [comment]
ugh, that test failures is basically:
```
E   -      location: sentry://project_debug_file/1
E   +      location: sentry://project_debug_file/12
```
How reliable is this number?  I created the snapshot from just executing the one test, but it seems more state is carried over between tests.  However current snapshots must already have this issue.  Do we just rely on the fact that all tests are always run in order?

**Comment by jan-auer** [comment]
The database IDs in tests are indeterministic. Ideally strip those out or interpolate with the identifiers that you get during fixture setup.

**Review by untitaker — DISMISSED** [review]

**Comment by flub** [comment]
ok, this is finally back into a shape that I think is fine.  please review :)

File: tests/symbolicator/test_payload_full.py [hunk 1]
```diff
@@ -185,3 +187,82 @@ def test_missing_debug_images(self):
...
+                    }
+                ]
+            },
+            "exception": {
+                "stacktrace": {
+                    "frames": [
+                        {
+                            "function": "<unknown>",
+                            "instruction_addr": "0x2a2a3d",
+                            "package": "C:\\projects\\breakpad-tools\\windows\\Release\\crash.exe",
```
Comment (jan-auer) [inline tests/symbolicator/test_payload_full.py:234]: If you like, you can further reduce this input payload by removing:
 - frame: function, package,
 - OS context
 - exception's thread id

File: tests/symbolicator/test_payload_full.py [hunk 2]
```diff
@@ -185,3 +187,82 @@ def test_missing_debug_images(self):
...
+            "timestamp": iso_format(before_now(seconds=1)),
+        }
+
+        with self.feature("organizations:images-loaded-v2"):
+            event = self.post_and_retrieve_event(event_data)
+        assert event.data["culprit"] == "main"
+
+        candidates = event.data["debug_meta"]["images"][0]["candidates"]
+        redact_location(candidates)
+        self.insta_snapshot(candidates)
```
Comment (jan-auer) [inline tests/symbolicator/test_payload_full.py:252]: You could also inspect the candidates list directly and check if the location is `f"sentry://project_debug_file/{debug_file.id}"`. You can get the debug file in the return value from `ProjectDebugFile.objects.create`.

**Review by jan-auer — APPROVED** [review]
Symbolicator tests have been fixed in `master`.

**Comment by flub** [comment]
Please keep alive, but thanks for the reminder.

File: tests/symbolicator/test_payload_full.py (hunk 2 above)
Comment (flub) [reply tests/symbolicator/test_payload_full.py:248]: I think I still want the insta snapshot for the other fields i think.  Not sure if the specific ID is important enough to test, let's not for now as that's less brittle, until this is known to be a problem.

**Pull request merged**